### DISCUSSION TITLE
Can I pick cells in mesh using mouse? How to?

### DISCUSSION THREAD

**Discussion by ZiguoAtGitHub** [marcomusy/vedo#751] [Q&A] [answered]
Dear Marco,
One more thing I should add that there maybe include different type meshes in the plot, such as polygons, lines or spheres etc.
Could you give me some hints?
Thank you!
🙂

**Comment by marcomusy** [comment, marked as answer]
you can use a callback function, eg
```python
from vedo import *

def callb(evt):
    msh = evt.actor
    if not msh:
        return
    pt = evt.picked3d
    idcell = msh.closest_point(pt, return_cell_id=True)
    msh.cellcolors[idcell] = [255,0,0,255] # red, opaque
    
m = Mesh(dataurl + "290.vtk")
m.decimate().smooth().compute_normals()
m.compute_quality().cmap("Blues", on="cells")

plt = Plotter()
plt.add_callback("mouse click", callb)
plt.show(m, m.labels("cellid"))
plt.close()
```
<img width="378" alt="Screenshot 2022-12-10 at 13 05 33" src="https://user-images.githubusercontent.com/32848391/206854278-1e62900b-00d5-41c2-aaf9-0b721002e964.png">

**Reply by ZiguoAtGitHub** [reply]
Hi Marco,
That Is What I Want!
Many thanks for your wonderful example.
🙂
Ziguo

**Comment by ZiguoAtGitHub** [comment]
Hi Marco @marcomusy ,
The example above can not run correctly today. The error is " builtins.AttributeError: 'Mesh' object has no attribute 'cellcolors' " when I pick a cell.
The script did run correctly before and nothing is changed with python and its' other settings of my pc.
I checked the online doc and found the var cellcolors is under vedo.pointcloud not vedo.mesh. Is that the reason?

Call for help~. 🙂
Thanks!

**Reply by marcomusy** [reply]
Hi, it still work fine for me!
maybe your environment has some problem (anaconda?)
Try
```
vedo --info
```

**Comment by ZiguoAtGitHub** [comment]
Hi
The script doesn't work both in my office pc and home. 
Here is information in my office pc. My home pc envs is python3.9+vedo 4.1+vtk9.2.2 (more information may be added later today.)
```
vedo version      : 2022.4.1.dev2   https://vedo.embl.es
vtk version       : 9.1.0
python version    : 3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:48:03) [MSC v.1928 64 bit (AMD64)]
python interpreter: C:\WPy64-38100\python-3.8.10.amd64\python.exe
vedo installation : C:\WPy64-38100\python-3.8.10.amd64\lib\site-packages\vedo
system            : Windows 10 nt AMD64
monitor           : info is unavailable. Try "pip install screeninfo".
```
I use Wing as my IDE. The problem so weird. I suppose the problem maybe disappear someday just like it appeared.
Thanks for your response.

**Comment by marcomusy** [comment]
You have an old version of vedo! try
```
pip install vedo -U
pip install vtk -U
```

**Reply by marcomusy** [reply]
..better try
```
pip install -U git+https://github.com/marcomusy/vedo.git
```

**Reply by ZiguoAtGitHub** [reply]
> You have an old version of vedo! try
> 
> ```
> pip install vedo -U
> pip install vtk -U
> ```

Thanks Marco. The problem is gone after upgrade vedo.
